### PR TITLE
init. simple camel case parser.

### PR DESCRIPTION
**Pull request by a-little-srdjan**
The code is geared towards readability (a simple push-down automata), and making it easy to change the token classes. People often have strong, differing, opinions on exactly how to slice a string :)

### CODE REVIEW

File: analysis/token_filters/camelcase_filter/parser.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	"github.com/blevesearch/bleve/analysis"
+)
+
+func buildTokenFromTerm(buffer []rune) *analysis.Token {
+	return &analysis.Token{
+		Term: buildTermFromRunes(buffer),
+	}
+}
+
+// TODO: Lifted from ngram_filter. Expose as public and re-use?
```
Comment (mschoch) [inline analysis/token_filters/camelcase_filter/parser.go]: This has been made public here:

https://github.com/blevesearch/bleve/blob/master/analysis/util.go#L37

**Comment by mschoch** [comment]
Just had that one comment, will merge now anyway, we can clean that up later.

**Comment by a-little-srdjan** [comment]
Bugger, always look at util :) I will make another pull request to clean it up. Thanks for the pointer

**Comment by mschoch** [comment]
Well, since you copied from ngram_filter, it sounds like we're not even using it throughout the codebase.  We should probably grep through and find similar cases.

**Comment by a-little-srdjan** [comment]
I've gone through the token filters. I found that the following filters can be refactored:
EdgeNgram
Ngram
CamelCase :)
Truncate (this one can benefit directly from util.TruncateRunes)

I can go ahead and start refactoring these filters?

**Comment by mschoch** [comment]
Yep, that would be awesome.

**Comment by a-little-srdjan** [comment]
Submitted as the pull request #386

**Pull request merged**